### DISCUSSION TITLE
s2e.so linking issues

### DISCUSSION THREAD

**Discussion by JosephNzKinyua** [S2E/s2e-env#498] [Q&A]
Hello, I am new to ubuntu, I am currently using Ubuntu 23.10 and I am building s2e manually, I first tried building using the s2e_env and it still produced this same error:

make[3]: Leaving directory '/home/jkinyua/Downloads/s2e/Build2/guest-tools32'
[ 42%] Linking C shared library s2e.so
[ 50%] Built target maze
make[3]: Entering directory '/home/jkinyua/Downloads/s2e/Build2/guest-tools32'
make[3]: Leaving directory '/home/jkinyua/Downloads/s2e/Build2/guest-tools32'
[ 57%] Built target quicksort
[ 64%] Built target vulnerabilities
make[3]: Leaving directory '/home/jkinyua/Downloads/s2e/Build2/guest-tools32'
/usr/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libelf.so when searching for -lelf
/usr/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libelf.a when searching for -lelf
/usr/bin/ld: skipping incompatible /usr/local/lib/libelf.so when searching for -lelf
/usr/bin/ld: skipping incompatible /usr/local/lib/libelf.a when searching for -lelf
/usr/bin/ld: cannot find -lelf: No such file or directory
/usr/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libelf.so when searching for -lelf
/usr/bin/ld: skipping incompatible /usr/local/lib/libelf.so when searching for -lelf
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
[ 71%] Built target cgccmd
make[3]: *** [linux/s2e.so/CMakeFiles/s2e.dir/build.make:209: linux/s2e.so/s2e.so] Error 1
make[3]: Leaving directory '/home/jkinyua/Downloads/s2e/Build2/guest-tools32'
make[2]: *** [CMakeFiles/Makefile2:432: linux/s2e.so/CMakeFiles/s2e.dir/all] Error 2
make[2]: Leaving directory '/home/jkinyua/Downloads/s2e/Build2/guest-tools32'
make[1]: *** [Makefile:136: all] Error 2
make[1]: Leaving directory '/home/jkinyua/Downloads/s2e/Build2/guest-tools32'
make: *** [/home/jkinyua/Downloads/s2e/Makefile:220: stamps/guest-tools32-make] Error 2

I need help resolving this issue:

**Comment by vitalych** [comment]
Hi, can you check if the 32-bit version of libelf is installed?
Note that S2E does not support 23.10 officially, only 22.04 LTS.

Vitaly

**Reply by JosephNzKinyua** [reply]
Hi, Thank you for your timely reply. I am not sure whether what I have is 32-bit. I installed it using this command:
sudo apt-get update
sudo apt-get install libelf-dev
